File: src/main/java/com/temelyan/pomoapp/repository/ProjectRepository.java
package com.temelyan.pomoapp.repository;

import com.temelyan.pomoapp.model.Project;

import java.time.LocalDate;
import java.util.Set;

public interface ProjectRepository {
    Project save(Project project, int userId);

    Set<Project> getAllForUserWithTasks(int userId);

    Set<Project> getAllForUserWithTasksAndPomos(int userId, LocalDate from, LocalDate to);

    Project getByIdWithUser(int projectId);
}

File: src/main/java/com/temelyan/pomoapp/repository/ProjectRepository.kt
package com.temelyan.pomoapp.repository

import com.temelyan.pomoapp.model.Project

import java.time.LocalDateTime

interface ProjectRepository {
    fun save(project: Project, userId: Int): Project

    fun getAllForUserWithTasks(userId: Int): Set<Project>

    fun getAllForUserWithTasksAndPomos(userId: Int, from: LocalDateTime, to: LocalDateTime): Set<Project>
}
